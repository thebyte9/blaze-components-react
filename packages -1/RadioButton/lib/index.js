"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
exports.__esModule = true;
var utils_1 = __importDefault(require("@blaze-react/utils"));
var react_1 = __importStar(require("react"));
var RadioButton = function (_a) {
    var onChange = _a.onChange, options = _a.options, required = _a.required, error = _a.error, validationMessage = _a.validationMessage, ErrorMessage = _a.utils.ErrorMessage, attrs = __rest(_a, ["onChange", "options", "required", "error", "validationMessage", "utils"]);
    var _b = react_1.useState({
        value: null
    }), selected = _b[0], setSelected = _b[1];
    var handleSelect = function (_a) {
        var event = _a.event, item = _a.item;
        if (item.disabled) {
            return;
        }
        setSelected(item);
        onChange({ event: event, selected: item });
    };
    return (react_1["default"].createElement(react_1.Fragment, null,
        required && react_1["default"].createElement("span", { className: "required" }),
        options.map(function (item) {
            var value = item.value, disabled = item.disabled, label = item.label, id = item.id;
            return (react_1["default"].createElement("div", { key: label, className: "form-field form-field--radio", onClick: function (event) { return handleSelect({ event: event, item: item }); }, role: "button" },
                react_1["default"].createElement("input", __assign({ readOnly: true, type: "radio", className: "form-radio", value: value, disabled: disabled, checked: value === selected.value, id: id }, attrs)),
                react_1["default"].createElement("label", { htmlFor: id }, label)));
        }),
        error && react_1["default"].createElement(ErrorMessage, { message: validationMessage })));
};
RadioButton.defaultProps = {
    error: false,
    onChange: function () {
        return;
    },
    options: [],
    required: false,
    validationMessage: "This field is required"
};
exports["default"] = utils_1["default"](RadioButton);
