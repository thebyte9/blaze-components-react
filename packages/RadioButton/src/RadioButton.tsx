import { ErrorMessage } from '@blaze-react/utils';
import React, { Fragment, FunctionComponent, useState } from 'react';

interface IErrorMessage {
  message: string | JSX.Element;
  icon?: string;
}

interface IRadioButtonProps {
  options: any[];
  required?: boolean;
  onChange: ({
    event,
    selected,
  }: {
    event: React.ChangeEvent<HTMLInputElement>;
    selected: string | Record<string, unknown> | [];
  }) => void;
  utils: {
    ErrorMessage: FunctionComponent<IErrorMessage>;
  };
  error?: boolean;
  validationMessage: string | JSX.Element;
}

const RadioButton: React.SFC<IRadioButtonProps> = ({
  onChange,
  options,
  required,
  error,
  validationMessage,
  ...attrs
}) => {
  const [selected, setSelected] = useState<{ value: string | any }>({
    value: null,
  });
  const handleSelect = ({ event, item }: { event: any; item: any }) => {
    if (item.disabled) {
      return;
    }
    setSelected(item);
    onChange({ event, selected: item });
  };
  return (
    <Fragment>
      {required && <span className="required" />}
      {options.map((item) => {
        const { value, disabled, label, id } = item;
        return (
          <div
            key={label}
            className="form-field form-field--radio"
            onClick={(event) => handleSelect({ event, item })}
            role="button"
          >
            <input
              readOnly
              type="radio"
              className="form-radio"
              value={value}
              disabled={disabled}
              checked={value === selected.value}
              id={id}
              {...attrs}
            />
            <label htmlFor={id}>{label}</label>
          </div>
        );
      })}
      {error && <ErrorMessage message={validationMessage} />}
    </Fragment>
  );
};

RadioButton.defaultProps = {
  error: false,
  onChange: (): void => {
    return;
  },
  options: [],
  required: false,
  validationMessage: 'This field is required',
};

export default RadioButton;
