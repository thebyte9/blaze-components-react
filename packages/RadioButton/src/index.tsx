import React, { Fragment, useState } from "react";
type RadioButtonProps = {
  options: any[],
  required?: boolean,
  onChange: (...args: any[]) => any
};
const RadioButton: React.SFC<RadioButtonProps> = ({
  onChange,
  options,
  required,
  ...attrs
}) => {
  const [selected, setSelected] = useState<{ value: string | any }>({ value: null });
  const handleSelect = ({ event, item }: { event: any, item: any }) => {
    if (item.disabled) return;
    setSelected(item);
    onChange({ event, selected: item });
  };
  return (
    <Fragment>
      {required && <span className="required" />}
      {options.map(item => {
        const { value, disabled, label, id } = item;
        return (
          <div
            key={label}
            className="form-field form-field--radio"
            onClick={event => handleSelect({ event, item })}
            role="button"
          >
            <input
              readOnly
              type="radio"
              className="form-radio"
              value={value}
              disabled={disabled}
              checked={value === selected.value}
              id={id}
              {...attrs}
            />
            <label htmlFor={id}>{label}</label>
          </div>
        );
      })}
    </Fragment>
  );
};
RadioButton.defaultProps = {
  options: [],
  required: false,
  onChange: () => { }
};
export default RadioButton;
