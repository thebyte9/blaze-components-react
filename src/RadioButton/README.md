
## Description

Radio button allows user to choose only one of the pre-defined options. Each radio button must be accompanied by a label describing the choice it represent.

## Usage


```js
const options = [
  { 
    label: 'A', 
    value: 1, 
    id: 'one' 
  },
  {
    label: 'B', 
    value: 2, 
    required: true, 
    id: 'two'
  },
  {
    label: 'C', 
    value: 3, 
    id: 'three'
  },
  {
    label: 'Disabled', 
    value: '', 
    disabled: true, 
    id: 'four'
  }
];

<RadioButton 
    required
    options={options} 
    onChange={({event, selected}) => {}} />
```

## API

##### RadioButton can receive a number of `props` as follow:


| NAME   | TYPE | DEFAULT | Description |
| :---  | :---:  | :---: | ------- |
| options | Array | [] | Array of objects |
| required | Boolean | false | Is a selection required? |
| onChange | Function | () => {} | returns callback with event and selecte option |

