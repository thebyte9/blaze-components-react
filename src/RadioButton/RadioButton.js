import React, { Fragment } from 'react';
import PropTypes from 'prop-types';

const RadioButton = ({
  onChange,
  options,
  required,
  ...attrs
}) => {
  const [selected, setSelected] = React.useState({});

  const handleSelect = ({ e, item }) => {
    if (item.disabled) return;
    setSelected(item);
    onChange({ e, selected: item });
  };

  return (
    <Fragment>
      {options.map((item) => {
        const {
          value,
          disabled,
          label
        } = item;

        return (
          <span
            key={label}
            className="form-field form-field--radio"
            onClick={e => handleSelect({ e, item })}
            role="button">
            <input
                readOnly
                type="radio"
                className="form-radio"
                value={value}
                disabled={disabled}
                checked={value === selected.value}
                {...attrs}
                />
            <label htmlFor={attrs.id}>{label}</label>
          </span>
        );
      })}
    </Fragment>
  );
};

RadioButton.propTypes = {
  options: PropTypes.array,
  onChange: PropTypes.func,
  required: PropTypes.bool
};

RadioButton.defaultProps = {
  options: {},
  required: false,
  onChange: () => {}
};

export default RadioButton;
