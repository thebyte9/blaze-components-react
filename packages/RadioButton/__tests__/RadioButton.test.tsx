import '@testing-library/jest-dom';

import { render, screen } from '@testing-library/react';

import RadioButton from '../src/RadioButton';
import React from 'react';
import userEvent from '@testing-library/user-event';

const options = [
  {
    label: 'Example',
    value: 1,
    id: 'one',
  },
  {
    label: 'I accept',
    value: 'accepted',
    required: true,
    id: 'two',
  },
  {
    label: 'Disabled',
    value: '',
    disabled: true,
    id: 'three',
  },
];

describe('RadioButton component', () => {
  test('should be defined xand renders correctly (snapshot)', () => {
    const { asFragment } = render(<RadioButton onChange={() => ({})} required options={options} />);
    expect(asFragment).toMatchSnapshot();
  });

  test('should active RadioButton on click', () => {
    render(<RadioButton onChange={() => ({})} options={options} />);
    userEvent.click(screen.getByLabelText(/I accept/i));
  });

  test("can't interact when RadioButton is disabled", () => {
    render(<RadioButton onChange={() => ({})} options={options} />);
    userEvent.click(screen.getByLabelText(/Disabled/i));
  });
});
