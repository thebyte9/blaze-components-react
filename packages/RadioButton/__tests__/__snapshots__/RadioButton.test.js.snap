// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioButton component should be defined and renders correctly (snapshot) 1`] = `ShallowWrapper {}`;
